services/sysinfo/server/sysinfo_linux.go

```diff
@@ -195,3 +196,16 @@ var getJournalRecordsAndSend = func(ctx context.Context, req *pb.JournalRequest,
 	}
 	return nil
 }
+
+func getTimeout(provided int64) time.Duration {
+	const maxTimeout = 30 * time.Second
+	const minTimeout = 2 * time.Second
+	timeout := time.Duration(provided) * time.Second
+	if timeout > maxTimeout {
+		timeout = maxTimeout
+	}
+	if timeout < minTimeout {
+		timeout = minTimeout
+	}
```

Comment:
Since this adjusts silently (without feedback to customers) values on server side I would consider it a defence in depth and have these ranges validated (with error messages) also in sanssh client.